frontend/components/CheatsheetForm.vue

```diff
@@ -0,0 +1,208 @@
+        <div v-for="(block, blockIndex) in blocks" :key="block.id">
+          <div class="field">
+            <FloatLabel>
+              <label>Section {{ blockIndex + 1 }}</label>
+              <InputText v-model="block.title" type="text" placeholder="Section Title" required />
+            </FloatLabel>
+          </div>
+
+          <!-- Contenu de la Section -->
+          <div v-for="(content, contentIndex) in block.contents" :key="content.id">
+            <div class="field">
+              <Editor v-model="content.values['fixed-column-id']" type="text" placeholder="Contenu" :style="{ height: '150px' }" />
+              <Button
+                type="button"
+                icon="pi pi-times"
+                severity="danger"
+                @click="removeContent(blockIndex, contentIndex)"
+              />
+            </div>
+          </div>
+
+          <!-- Ajouter du Contenu -->
+          <div class="field">
+            <Button
+              type="button"
+              severity="info"
+              icon="pi pi-plus"
+              label="Add a content line"
+              @click="addContent(blockIndex)"
+            />
+
+            <!-- Supprimer la Section -->
+            <Button
+              type="button"
+              severity="danger"
+              label="Delete section"
+              icon="pi pi-times"
+              @click="removeBlock(blockIndex)"
+            />
+          </div>
+        </div>
+
+        <!-- Ajouter une Section -->
+        <div class="field">
+          <Button
+            type="button"
+            severity="info"
+            label="Add a section"
+            icon="pi pi-folder-plus"
+            @click="addBlock"
+          />
+
+          <!-- Soumettre la Cheatsheet -->
+          <Button
+            type="submit"
+            severity="success"
+            :loading="isLoading"
+            label="Publish"
+            icon="pi pi-check"
+          />
+        </div>
+      </VForm>
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { ref, onMounted } from 'vue'
+import { v4 as uuidv4 } from 'uuid'
+import axios from 'axios'
+import * as yup from 'yup';
+import { Field as VField, Form as VForm } from 'vee-validate';
+
+// Import PrimeVue Components (PrimeVue v4)
+import Button from 'primevue/button'
+import InputText from 'primevue/inputtext'
+import Editor from 'primevue/editor'
+import type { Category, Block, Content } from '../types/cheatsheet'
+
+const config = useRuntimeConfig();
+
+// State
+const cheatsheetTitle = ref<string>('')
+const blocks = ref<Block[]>([])
+const isLoading = ref(false)
+
+// Pour les catégories
+const categories = ref<Category[]>([])
+const selectedCategory = ref<string>('') // IRI de la catégorie sélectionnée
+
+const cheatsheetTitleSchema = yup
+  .string()
+  .min(3, 'The title must contain at least 3 characters')
+  .max(100, 'The title must not exceed 100 characters.')
+  .required('Title is required');
+
```

Comment:
Something like this: 
```js
const { values, handleSubmit } = useForm({
  validationSchema: toTypedSchema(
    object({
      email: string().min(1, 'required'),
      password: string().min(1, 'required'),
      name: string().optional(),
    }),
  ),
});
```

https://vee-validate.logaretm.com/v4/guide/composition-api/typed-schema#zod